### DISCUSSION TITLE
Get theme names from SCSS list variable and use it for dynamic color palette names

### DISCUSSION THREAD

**Discussion by micobarac** [tailwindlabs/tailwindcss#13075] [General]
I have color palettes defined inside SCSS file, but I want them to be available to Tailwind, too. Colors are mapped fine by using variables, but getting and parsing theme names from SCSS to dynamically build theme colors doesn't work...

```css
--themes: apple, space, orange, gray, mid-gray, soft-gray, almost-white, sky, neon;
```

```js
  const themes = 'var(--themes)'.split(',').map((theme) => theme.trim());
  console.log(themes);
```

The console log:

`[ 'var(--themes)' ]`

Full code:

```js
const getColors = () => {
  const colors = {};
  const shades = [50, 100, 200, 300, 400, 500, 600, 700, 800, 900, 950];

  const themes = 'var(--themes)'.split(',').map((theme) => theme.trim());

  themes.forEach((theme) => {
    const tints = {};

    shades.forEach((shade) => {
      tints[shade] = `rgb(var(--${theme}-${shade}) / <alpha-value>)`;
    });

    colors[theme] = tints;
  });

  return colors;
};
```

I know that like this, the CSS variable is not parsed, it is just a string, so I need a suggestion how to get it right.

**Comment by wongjn** [comment]
You could consider replacing the `'var(--themes)'` parsing with the list of themes as an array:
```diff
-const themes = 'var(--themes)'.split(',').map((theme) => theme.trim());
+const themes = [
+  'apple',
+  'space',
+  'orange',
+  'gray',
+  'mid-gray',
+  'soft-gray',
+  'almost-white',
+  'sky',
+  'neo',
+];
```

**Reply by micobarac** [reply]
I don't want the theme names defined inside js file. I need to get them as SCSS variables.

**Reply by wongjn** [reply]
You can define them in SCSS using [a list](https://sass-lang.com/documentation/values/lists/) like:

```scss
$themes: 'apple', 'space', 'orange', 'gray', 'mid-gray', 'soft-gray', 'almost-white', 'sky', 'neon';
```

**Reply by micobarac** [reply]
It is defined like it, but I cannot get those values inside tailwind.config.js.

**Reply by wongjn** [reply]
You could consider defining them like https://github.com/tailwindlabs/tailwindcss/discussions/13075#discussioncomment-8663245

**Reply by micobarac** [reply]
You are not helpful...

**Reply by micobarac** [reply]
I don't want to define theme names inside Tailwind configuration. I want to get it from SCSS variables defined in SCSS file.

**Reply by wongjn** [reply]
I'm sorry that you feel that way. Could you elaborate more about how the solution is not helpful to you please?

**Reply by wongjn** [reply]
> I don't want to define theme names inside Tailwind configuration. I want to get it from SCSS variables defined in SCSS file.

I see!

You could read the file and then parse the content manually, like:

```js
const fs = require('node:fs');

const fileContent = fs.readFileSync('./path/to/file.scss', 'utf8');
const themesValue = /\$theme:(.+?);/.exec(fileContent);
const themes = // Extract values from `themesValue` here.
```

**Reply by micobarac** [reply]
Not working :(

**Reply by wongjn** [reply]
You'll need to adapt it to make it work for your project.
